### PR TITLE
Added{Join Meeting}:-Added The meeting files.

### PR DESCRIPTION
**Pull request by RAM1232**
## Description
This pull request implements the join meeting functionalities into the application.

## Jira Ticket
[SPARK-526690](https://jira-eng-gpk2.cisco.com/jira/browse/SPARK-526690)

## Demo Recording
https://github.com/mkesavan13/webex-lyte/assets/46831391/5c765271-4755-4653-8b83-db5db3fbe8a5

## Changes
- Added meeting.html, meeting.css, meeting.js.
- Implemented Join Meeting Functionalities.
- Implementing Functionalities like Start/Stop, Mute/Unmute, Leave.
- Updated UI including Bootstrap Modal view.

### CODE REVIEW

File: meeting.css
Comment (mkesavan13) [inline meeting.css]: Please ensure to keep all the variable names starting with a lower case alphabet

File: meeting.html
Comment (mkesavan13) [inline meeting.html]: Let's please avoid any inline css that we add. Inline CSS aren't preferred. It's always better to write these rules under a class or ID selector

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: Let's not commit any access token values going forward. These are public repositories and having an access token like that could sabotage your account

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: While you have added these event listeners on `DOMContentLoaded` why is `leaveMeetingButton` gets an event callback added differently in line number 94?

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: We should instead be using `createVideoStream` and `createAudioStream` methods from `mediaHelpers`

Please check the [webex-js-sdk-meeting-demo webex.js:138](https://github.com/WebexSamples/webex-js-sdk-meeting-demo/blob/main/webex.js#L138) for reference.

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: The SDK doesn't use/emit track events anymore. Please refactor this to use streams.

File: meeting.html
Comment (mkesavan13) [inline meeting.html]: This is a V1 SDK's CDN URL. We are right now in V3 as we all know.

Please update the version and try to setup meetings accordingly.

**Review by mkesavan13 — CHANGES_REQUESTED** [review]

File: meeting.js
Comment (RAM1232) [reply meeting.js]: Kesava these were loaded when the meeting occurred and the leave meeting button occurs to showcase the current so I made it when the document is loaded and join meeting comes under div one so I included it we can make it directly in the function itself for better and differentiate with the parameters are loaded before and after meeting.

**Comment by RAM1232** [comment]
@mkesavan13 Hi Kesava I have updated the files. Please Review it

File: meeting.html
Comment (mkesavan13) [inline meeting.html]: What is this image? Why do we need it?

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: Why are we setting specific width and height here?

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: Let us try to get rid of jquery and use Browsers DOM methods like `document.getElementById`

This comment is applicable wherever we have used jQuery `$()`

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: Instead of replacing the whole HTML, we should simply be replacing the class name

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: Same here. Applicable everywhere

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: Why are we adjusting padding here? Shouldn't padding be the same throughout?

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: Same comment here. Applicable wherever we have made inline CSS change

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: Creating the audio context is optional. We do not need that. Let's create noise reduction effect without a custom audio context and let the WxEffects library handle it

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: - How does it determine the mode is background blur?
- Can we also support Video Mirroring as part of this? I see that it is possible with the wiki [here](https://github.com/webex/webex-js-sdk/wiki/Streams-and-Effects#virtual-background-effect).

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: What is the purpose of handling permission change here?

**Review by mkesavan13 — CHANGES_REQUESTED** [review]

**Comment by mkesavan13** [comment]
Also @RAM1232,
We should have all these changes in the `index.html` file itself right? We aren't redirecting to a new page to join a meeting. 

Therefore please pull @Parimala032's changes from main branch, make your changes into a single file and then we are good to go. Let's have a demo recording added to the PR before we merge it.

File: meeting.html
Comment (RAM1232) [reply meeting.html]: Kesava This is regarding the image not related to button  it was related to the video tag when we stop the local Video it will display instead of icon we can use image url  to show that video is stop

File: meeting.js
Comment (RAM1232) [reply meeting.js]: These are the constraits to give the video quality of the localvideo.So we give the constraits for better video quality

File: meeting.js
Comment (RAM1232) [reply meeting.js]: This is modelview bootstrap command so when we use bootstrap command we use jquery

File: meeting.js
Comment (RAM1232) [reply meeting.js]: this is seperate icon to mute and unmute so if we give the class name directly the then it will not replicate the icon switches like we are using different icons for mute,unmute start,stop .So switching from one icon to another icon we use innerHtml rather than giving class name

File: meeting.js
Comment (RAM1232) [reply meeting.js]: this is seperate icon to mute and unmute so if we give the class name directly the then it will not replicate the icon switches like we are using different icons for mute,unmute start,stop .So switching from one icon to another icon we use innerHtml rather than giving class name

File: meeting.js
Comment (RAM1232) [reply meeting.js]: No Kesava when we use different icon for UnMute The padding was different due to icons size.I have tried by changing the icon size but it did not work so i had use the padding.

File: meeting.js
Comment (RAM1232) [reply meeting.js]: No Kesava when we use different icon for UnMute The padding was different due to icons size.I have tried by changing the icon size but it did not work so i had use the padding.

File: meeting.js
Comment (RAM1232) [reply meeting.js]: OK Kesava Need to look through it

File: meeting.js
Comment (RAM1232) [reply meeting.js]: When the Camera Permission and microphone permission was off it will automatically off the local video and microphone so we can on the microphone and camera instead of another button called load camera

File: meeting.js
Comment (RAM1232) [reply meeting.js]: You have said to keep background as blur by making Video Mirroring we are flipping image output . we can keep the background as blur. in the documentation given when we not given any parameters in the function the background by default taken as blur we can adjust the blur by adding blur strength.

File: meeting.html
Comment (RAM1232) [reply meeting.html]: Screenshot 2024-06-18 123410
Here you can observe the black Video Tag that was the image url instead of using icon

File: meeting.js
Comment (RAM1232) [reply meeting.js]: In the documentation also the have use the constraits like width and height for better quality of local Video Image

File: meeting.js
Comment (RAM1232) [reply meeting.js]: That was by default modal-view format.When we use bootstrap modal view we should use jquerry to show the modal view

File: meeting.js
Comment (RAM1232) [reply meeting.js]: Like we have to remove existing class and add the related class so i had updated with the innerHTML

File: meeting.js
Comment (RAM1232) [reply meeting.js]: vbgEffect = await webex.meetings.createVirtualBackgroundEffect(); when we not give any parameters by default it will take the blur background

File: meeting.js
Comment (mkesavan13) [reply meeting.js]: Have you explored the [javascript classList](https://www.w3schools.com/jsref/prop_element_classlist.asp) property?

File: meeting.js
Comment (mkesavan13) [reply meeting.js]: The font-awesome icons should have an equal sized icon set. Even if we are adjusting padding, it should be done from CSS and not from Javascript.

Better to write CSS based on the icon class names and adding padding there so it adjusts automatically

File: meeting.js
Comment (mkesavan13) [reply meeting.js]: The font-awesome icons should have an equal sized icon set. Even if we are adjusting padding, it should be done from CSS and not from Javascript.

Better to write CSS based on the icon class names and adding padding there so it adjusts automatically

File: meeting.js
Comment (mkesavan13) [reply meeting.js]: Makes sense. Thank you.

We can just have a checkbox to mirror video. If it is checked, we can mirror and if it is unchecked we can remove mirror from options.

File: meeting.js
Comment (mkesavan13) [reply meeting.js]: Ideally, start video would load camera from inside. In the Kitchen Sink app, we have a separate button to indicate there's a separate API. 

If the permission changes, the page needs to be reloaded anyway, right?

File: meeting.js
Comment (RAM1232) [reply meeting.js]: By using DOM methods we cannot achive the modal view as you can see the method use modal method.

File: meeting.html
Comment (mkesavan13) [reply meeting.html]: I don't think we need an image. Let us change this to font awesome icons

File: meeting.js
Comment (mkesavan13) [reply meeting.js]: What happens if we do not give this width and height? I think they are optional. Do we really need to constrain it with a hardcoded resolution here?

File: meeting.js
Comment (mkesavan13) [reply meeting.js]: Oh. So, the bootstrap framework mandates jquery usage? Got it.

File: meeting.js
Comment (mkesavan13) [reply meeting.js]: Have you explored the [javascript classList](https://www.w3schools.com/jsref/prop_element_classlist.asp) property?

File: meeting.html
Comment (RAM1232) [reply meeting.html]: By using Font Awesome Icons We cannot Set Size Of Icon

File: meeting.js
Comment (RAM1232) [reply meeting.js]: The resolution will change when we give constraints So I had given Constraints That was the ratio For 1080 p 720p like that . Refer the Documentation and also Sample App Reference uses constraints

File: meeting.js
Comment (RAM1232) [reply meeting.js]: Yes I had Explored Class List Property And applied But it did not Works As Expected Compare to innerHtml

File: meeting.js
Comment (RAM1232) [reply meeting.js]: Yes I had Explored Class List Property And applied But it did not Works As Expected Compare to innerHtml

File: meeting.js
Comment (RAM1232) [reply meeting.js]: Like All The Styles Were perfect but The Icon left padding didn't work So I had added the js .So If needed I can add Seperate CSS And add it as an Class list

File: meeting.js
Comment (RAM1232) [reply meeting.js]: Like All The Styles Were perfect but The Icon left padding didn't work So I had added the js .So If needed I can add Seperate CSS And add it as an Class list

File: meeting.js
Comment (RAM1232) [reply meeting.js]: Like I had set mirror default as true if we add check box where to add

File: meeting.js
Comment (RAM1232) [reply meeting.js]: Like It will show reloaded option but what if the user in the meeting middle so I had add these permission also

File: meeting.html
Comment (RAM1232) [reply meeting.html]: Updated

File: meeting.js
Comment (RAM1232) [reply meeting.js]: Updated

File: meeting.js
Comment (RAM1232) [reply meeting.js]: I had applied the through class list it did not expect from the result as innerHTML.And why we use class list instead of innerHtml for the code.

File: meeting.js
Comment (RAM1232) [reply meeting.js]: I had applied the through class list it did not expect from the result as innerHTML.And why we use class list instead of innerHtml for the code.

File: meeting.js
Comment (RAM1232) [reply meeting.js]: Updated

File: meeting.js
Comment (RAM1232) [reply meeting.js]: Instead of having checkbox can we implemented internally through VBG.

**Comment by RAM1232** [comment]
I think so.But After merging my PR i can do in a seperate branch such that i can able to connect two pages so I had not rebase with origin.So Firstly i was working on remaining comments if any comments were not there i will rebase with original branch and interlinked pages

File: meeting.js
Comment (mkesavan13) [inline meeting.js]: Why aren't we having `audioEnabled` and `videoEnabled` here?

Comment (RAM1232) [reply meeting.js]: Updated

**Comment by RAM1232** [comment]
@mkesavan13 I Updated the Code Based on Recent comments. Please review.

**Comment by RAM1232** [comment]
@mkesavan13 I rebased the code with the main branch and integrated two codes into a single file (index.html, app.js, meeting.css, and style.css). Please review it. After reviewing, we can test the code

**Comment by mkesavan13** [comment]
@Parimala032 @RAM1232 

Kindly take a look at the following images and update the UI accordingly. They are minor but worth changing.

Screenshot 2024-06-20 at 7 38 57 AM

---

Screenshot 2024-06-20 at 7 34 26 AM

---

Screenshot 2024-06-20 at 7 28 39 AM

File: meeting.js
Comment (mkesavan13) [reply meeting.js]: Had a long 1:1 conversation on why.

innerHTML removes the whole element and pushes new element on the UI causing a re-render while classList only switches the CSS rules that are already loaded and parsed in the DOM. That is why it is good to prefer classList than switching HTML

**Review by mkesavan13 — COMMENTED** [review]
The PR looks good to me. Will approve and merge once the recent comment on the demo is addressed. 

Once the changes are done, please post a new recording of the same

**Comment by RAM1232** [comment]
Kesava Regarding First Image We Cannot Reduce The gap between the Button And Screen as Normally It was Fine But When The Remote Person Screen Share It will take Screen Based On that The Remote Screen Was Adjusted.Remaining Button Names I will Updated.For the Button Color if it will black and white we cannot see black color instead we can use dark grey and white

**Review by mkesavan13 — APPROVED** [review]
One final review comment. Approving the PR.

Screenshot 2024-06-24 at 11 54 00 AM

**Comment by RAM1232** [comment]
@mkesavan13 Hi Kesava, I updated the code based on the recent comments.
Screenshot 2024-06-24 124302
If it is fine, then we can go ahead.

**Review by mkesavan13 — APPROVED** [review]

**Pull request merged**
